src/main/java/teammates/common/datatransfer/FeedbackConstantSumQuestionDetails.java

```diff
@@ -377,34 +377,34 @@ public String getQuestionResultStatisticsHtml(
                 String name = bundle.getNameForEmail(participantIdentifier);
                 String teamName = bundle.getTeamNameForEmail(participantIdentifier);
                 
-                fragments += FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_STATS_RECIPIENTFRAGMENT,
+                fragments.append(FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_STATS_RECIPIENTFRAGMENT,
                         "${constSumOptionValue}",  Sanitizer.sanitizeForHtml(name),
                         "${team}", Sanitizer.sanitizeForHtml(teamName),
                         "${pointsReceived}", pointsReceived,
-                        "${averagePoints}", df.format(average));
+                        "${averagePoints}", df.format(average)));
             
             } else {
                 String option = options.get(Integer.parseInt(entry.getKey()));
                 
-                fragments += FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_STATS_OPTIONFRAGMENT,
+                fragments.append(FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_STATS_OPTIONFRAGMENT,
                                     "${constSumOptionValue}",  Sanitizer.sanitizeForHtml(option),
                                     "${pointsReceived}", pointsReceived,
-                                    "${averagePoints}", df.format(average));
+                                    "${averagePoints}", df.format(average)));
             }
         }
         
         if (distributeToRecipients) {
-            html = FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_RECIPIENT_STATS,
+            html.append(FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_RECIPIENT_STATS,
                     "${optionRecipientDisplayName}", "Recipient",
-                    "${fragments}", fragments);
+                    "${fragments}", fragments.toString()));
         } else {
-            html = FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_OPTION_STATS,
+            html.append(FeedbackQuestionFormTemplates.populateTemplate(FeedbackQuestionFormTemplates.CONSTSUM_RESULT_OPTION_STATS,
                     "${optionRecipientDisplayName}", "Option",
-                    "${fragments}", fragments);
+                    "${fragments}", fragments.toString()));
         }
         
         
-        return html;
+        return html.toString();
```

Comment:
I was actually delaying fixing all of these until adding in the relevant PMD rule (UnnecessaryLocalBeforeReturn?).